app/src/main/java/kr/ac/kaist/team888/util/FeatureController.java

```diff
@@ -0,0 +1,79 @@
+package kr.ac.kaist.team888.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class FeatureController {
+  
+  /**
+   * Interface for listening change of hangul features.
+   *
+   * <p> If some feature(weight, curve, and etc)'s value are change,
+   * {@link FeatureController} calls {@link #onFeatureChange()}.
+   *
+   */
+  public interface OnFeatureChangeListener {
+    /**
+     * Method that will be executed on feature's value change.
+     */
+    void onFeatureChange();
+
+    /**
+     * Priority of listener.
+     *
+     * <p> The lower priority executed first.
+     *
+     * @return priority as integer
+     */
+    int getPriority();
+  }
+
+  private static class Singleton {
+    private static final FeatureController instance = new FeatureController();
+  }
+
+  /**
+   * Getter of singleton instance.
+   *
+   * @return singleton instance
+   */
+  public static FeatureController getInstance() {
+    return FeatureController.Singleton.instance;
+  }
+
+  private ArrayList<OnFeatureChangeListener> listeners;
+
+  private FeatureController() {
+    listeners = new ArrayList<>();
+  }
+
+  /**
+   * Register {@link OnFeatureChangeListener}.
+   *
+   * @param listener listener to register
+   */
+  public void registerOnFeatureChangeListener(OnFeatureChangeListener listener) {
+    listeners.add(listener);
+    Collections.sort(listeners, new Comparator<OnFeatureChangeListener>() {
+      @Override
+      public int compare(OnFeatureChangeListener o1, OnFeatureChangeListener o2) {
+        if (o1.getPriority() > o2.getPriority()) {
+          return 1;
+        } else if (o1.getPriority() == o2.getPriority()) {
+          return 0;
+        } else {
+          return -1;
+        }
```

Comment:
What about
```java
return o1.getPriority() - o2.getPriority()
```
instead of the current multiline codes?
```java
if (o1.getPriority() > o2.getPriority()) {
  return 1;
} else if (o1.getPriority() == o2.getPriority()) {
  return 0;
} else {
  return -1;
}
```

https://docs.oracle.com/javase/7/docs/api/java/util/Comparator.html#compare(T,%20T) says that

> Compares its two arguments for order. Returns a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the second.

so doesn't matter if the return value is not in `[-1, 0, 1]`.